API/src/main/java/de/brandwatch/minianalytics/api/service/MentionService.java

```diff
@@ -0,0 +1,74 @@
+package de.brandwatch.minianalytics.api.service;
+
+import com.google.gson.Gson;
+import de.brandwatch.minianalytics.api.solr.model.Mention;
+import de.brandwatch.minianalytics.api.solr.repository.MentionRepository;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.time.*;
+import java.time.format.DateTimeFormatter;
+import java.time.temporal.TemporalAccessor;
+import java.util.List;
+
+@Component
+public class MentionService {
+
+    private static final Logger logger = LoggerFactory.getLogger(MentionService.class);
+
+    private final MentionRepository mentionRepository;
+
+    private static final Gson gson = new Gson();
+
+    @Autowired
+    public MentionService(MentionRepository mentionRepository) {
+        this.mentionRepository = mentionRepository;
+    }
+
+    public List<Mention> getMentionsFromQueryID(String queryID, String startDate, String endDate) {
+
+        StringBuilder sb = new StringBuilder();
+        sb.append("[");
+
+        if(!endDate.equals("") && !startDate.equals("")){
+            if(!validateDates(convertStringDateToInstant(startDate), convertStringDateToInstant(endDate))){
+                throw new DateTimeException("Dates are not in the right order");
+            }
+        }
+
+        if (startDate.equals("")) {
+            sb.append("*");
+        } else {
+            sb.append(convertStringDateToInstant(startDate));
+        }
+
+        sb.append(" TO ");
+
+        if (endDate.equals("")) {
+            sb.append("*");
+        } else {
+            sb.append(convertStringDateToInstant(endDate));
+        }
+
+        sb.append("]");
+
+        return mentionRepository.findMentionsAfterDate(Long.parseLong(queryID), sb.toString());
+    }
+
+    private Instant convertStringDateToInstant(String date) {
```

Comment:
I feel like I have commented on this method already in another PR, no? Anyway, there's multiple things going on here. For example, there is no need to  create a date time formatter for every method invocation. A bean will be just fine. Also, the line that adds "00:00:00" is very funky - why not accept only dates in the correct format in the first place? And then there is the conversion. You convert from LocalDateTime to ZonedDateTime to Instant. Shouldn't the controller just accept a date object of some kind directly? https://www.baeldung.com/spring-date-parameters
It could maybe accept LocalDateTime or ZonedDateTime, and you convert to Instant accordingly in the controller, and pass that down to the service.

Reply:
I had something similar in this https://github.com/Max-Leopold/mini-analytics/pull/22. We didn't finish our conversation there.

Reply:
> https://www.baeldung.com/spring-date-parameters
Didn't know this was possible. Will change.